textpipe/doc.py

```diff
@@ -26,64 +37,70 @@ def __init__(self, raw, language=None, hint_language='en'):
         _language: 2-letter code for the language of the text
         _hint_language: language you expect your text to be
         _clean_text: string containing the cleaned text
-        _spacy_nlps: dictionary containing a spacy language module
-        _spacy_doc: dictionary containing an instance of a spacy doc
+        _spacy_nlps: dictionary containing spacy language modules for languages/operations
+        _spacy_docs: dictionary containing instances of a spacy docs
         _text_stats: dictionary containing an instance of textacy textstats
         """
 
         self.raw = raw
         self.is_detected_language = language is None
         self._language = language
+        self.allowed_languages = allowed_languages
         self._hint_language = hint_language
         self._clean_text = None
-        self._spacy_nlps = {}
-        self._spacy_doc = {}
+        self.spacy_nlps = spacy_nlps
+        self._spacy_docs = {}
         self._text_stats = {}
+        # If used without Pipeline, set the spacy default language modules:
```

Comment:
or if there's no spacy nlps defined in the pipeline?

Reply:
Good point, I didn't think of the option to only load the default models in the Doc and the custom models in the Pipeline. Will commit a solution where only the custom models are loaded in the Pipeline, and the Doc adds the needed default language to the Pipeline spacy_nlps-object  on the go.